### ISSUE TITLE
Lake and river related issues 

### ISSUE THREAD

**Issue by WBMol** [openmaptiles/openmaptiles#320]
There are a couple of visual complications when dealing with water and lakes, at least in the Netherlands. Though these are mostly specific to our use case, I think more people would benefit from possible solutions.

**Zoom inconsistency**
First off, water is either 'ocean' or 'lake', and its classification is inconsistent among zoom levels. A related issue can be found here: https://github.com/openmaptiles/openmaptiles/issues/310

A prominent example can be found in the Zeeland province (south west NL). Zoom 0-8 shows this: 
![image](https://user-images.githubusercontent.com/13695103/29509441-935f6d8c-8658-11e7-99dd-26d7b74ae40c.png) where only the ocean and a custom lake (IJselmeer) has been coloured blue. 

Zoom 9+ turns the third from bottom ocean part into a lake, or in fact, the ocean part disappears. The lake polygons are present at lower zooms already (more or less duplicating the ocean): 
![image](https://user-images.githubusercontent.com/13695103/29509548-f07c99ea-8658-11e7-9fba-d62649066287.png)

**Scale rank of 'lakes'**
This would ordinarily not pose a problem, as you can use lake data to colour that part blue again. The problem here is, however, that a large lake like IJsselmeer can not be seperated from the smallest ponds and water bodies. And the Netherlands is full of the latter, which adds distractions at zooms 9+. Idealy, some sort of scale rank metadata to optionally filter these out would help a lot. Or, maybe even better, label ditches and ponds as such, instead of 'lakes'. 

An example of our base layer with the 'lakes' enabled between Amsterdam and Utrecht: 
![image](https://user-images.githubusercontent.com/13695103/29510014-9457e424-865a-11e7-9396-f7edf0a8b9eb.png)

Our current workaround is manually adding polygon data of lakes. This introduces new problems of course, like minor dataset inconsistency or missing lakes depending on the source dataset, resolution/quality issues etc. 

**Names of lakes**
Large lakes have no name attributes, though they would be nice to have. 

**Interrupted rivers**
While on the topic of water, I'd like to mention how some rivers are interrupted. Here an example of the city of Utrecht, which has a nice canal network. At lower zooms, there are gaps. Example at zoom 12 vs 13:
![image](https://user-images.githubusercontent.com/13695103/29510803-65e59f34-865d-11e7-9c7d-57b75208073a.png)
![image](https://user-images.githubusercontent.com/13695103/29510836-7f8896a8-865d-11e7-88ba-29b735784460.png)

I hope my examples are clear enough. If not, let me know.

**Comment by jirik** [comment]
@WBMol It all sounds quite reasonable, pull request is welcome.

Names of lakes: these are already presented in water_name layer.

**Comment by frodrigo** [comment]
I agree with the fact that too many thing finish with class=lake, from bay to pond.

**Comment by ZeLonewolf** [comment]
By processing the `water=*` tag, a scale ranking for water bodies could be as follows:

**Lowest rank**
* `water=basin`
* `water=stream`
* `water=wastewater`
* `water=ditch`

**Middle rank**
* `water=pond` (`natural=water` with no `water=*` classified as `pond`)
* `water=oxbow`
* `water=canal`

**Highest rank**
* `water=lagoon`
* `water=lake`
* `water=reservoir`

In addition, we could apply a more robust filtering for size, so that objects that are only a few pixels are generalized out.

I am interested to understand whether any style would care to differentiate between these types of `water=*` bodies, or if it's sufficient to keep the broad categories of `lake`/`river`/`ocean` and simply compute a rank column.  As an initial step before breaking changes in 4.0, I would propose encoding the water bodies above into `lake` or `river` and computing a scale value.  If there is a desire to have more specific water body rendering, we could still break up `lake` or `river` into more specific classes in 4.0 as a breaking change.

**Comment by farfromrefug** [comment]
I personally changed the water to return the true class and not almost always lake. I also added fountain btw.
It does take a little bit more space but for me it is important. The reason is that i changed water to also return the name of water bodies. So i use the class mostly to decide text size and when to show the name.
But it is only really  necessary in my case.
As for the scaling ranking you are talking about showing data  a zoom levels based on that ranking?
I personally now uses `way_pixels >0` or `zoom >=14`. That way i only show bodies of water when necessary and i dont have flashing effects od body of water sudenly appearing.

**Comment by ZeLonewolf** [comment]
@farfromrefug, I am looking forward to your `way_pixels` PR getting merged, as I think it might help with this issue.  Perhaps we might consider in general, or for certain classes, making the filter `way_pixels>N` where `N` is some number greater than zero which provides more aggressive filtering of insignificant water objects.

Thinking about this more and from your comments, maybe all we really need is to support the various `water=*` classes (in `water_name` in particular for labeling) combined increasing the cutoff for pixel size.
